cc/system_library.bzl

```diff
@@ -0,0 +1,448 @@
+    includes_param = "includes = {},".format(str(include_subdirs.keys()))
+
+    deps_names = []
+    for dep in deps:
+        dep_name = repr("@" + dep)
+        deps_names.append(dep_name)
+    deps_param = "deps = [{}],".format(",".join(deps_names))
+
+    link_hdrs_command = "mkdir -p $(RULEDIR)/remote \n"
+    remote_hdrs = []
+    for path, hdr in zip(hdr_paths, hdr_names):
+        remote_hdr = "remote/" + hdr
+        remote_hdrs.append(remote_hdr)
+        link_hdrs_command += "cp {path} $(RULEDIR)/{hdr}\n ".format(
+            path = path,
+            hdr = remote_hdr,
+        )
+
+    link_remote_static_lib_genrule = ""
+    link_remote_shared_lib_genrule = ""
+    remote_static_library_param = ""
+    remote_shared_library_param = ""
+    static_library_param = ""
+    shared_library_param = ""
+
+    if static_lib_path:
+        repo_ctx.symlink(static_lib_path, static_lib_name)
+        static_library_param = "static_library = \"{}\",".format(
+            static_lib_name,
+        )
+        remote_static_library = "remote/" + static_lib_name
+        link_library_command = \
+            "mkdir -p $(RULEDIR)/remote && cp {path} $(RULEDIR)/{lib}".format(
+                path = static_lib_path,
+                lib = remote_static_library,
+            )
+        remote_static_library_param = \
+            "static_library = \"remote_link_static_library\","
+        link_remote_static_lib_genrule = \
+            """
+genrule(
+     name = "remote_link_static_library",
+     outs = ["{remote_static_library}"],
+     cmd = {link_library_command}
+)
+""".format(
+                link_library_command = repr(link_library_command),
+                remote_static_library = remote_static_library,
+            )
+
+    if shared_lib_path:
+        repo_ctx.symlink(shared_lib_path, shared_lib_name)
+        shared_library_param = \
+            "shared_library = \"{}\",".format(shared_lib_name)
+        remote_shared_library = "remote/" + shared_lib_name
+        link_library_command = \
+            "mkdir -p $(RULEDIR)/remote && cp {path} $(RULEDIR)/{lib}".format(
+                path = shared_lib_path,
+                lib = remote_shared_library,
+            )
+        remote_shared_library_param = \
+            "shared_library = \"remote_link_shared_library\","
+        link_remote_shared_lib_genrule = \
+            """
+genrule(
+        name = "remote_link_shared_library",
+        outs = ["{remote_shared_library}"],
+        cmd = {link_library_command}
+)
+""".format(
+                link_library_command = repr(link_library_command),
+                remote_shared_library = remote_shared_library,
+            )
+
+    repo_ctx.file(
+        "BUILD",
+        executable = False,
+        content =
+            """
+load("@bazel_tools//tools/build_defs/cc:cc_import.bzl", "cc_import")
+cc_import(
+    name = "local_includes",
+    {static_library}
+    {shared_library}
+    {hdrs}
+    {deps}
+    {includes}
+)
+
+genrule(
```

Comment:
I know it's hard to test RBE. But if you can test the genrules themselves, let's do that, at least we get some signal.